napi_from_js.md

```diff
@@ -0,0 +1,367 @@
+representing the arguments to the function. If there are more arguments
+than the provided argument count, then only the requested number of 
+arguments are copied. If there are fewer arguments provided than 
+claimed, the rest of `argv` is filled with `napi_value` values that 
+represent `undefined`
+- `[out] result`: The JavaScript object representing the return 
+value from the function call
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This method allows you to call a JavaScript function object from your native 
+add-on. This is the primary mechanism of calling back *from* the add-on's
+native code *into* JavaScript.
+
+A sample use case might look as follows. If you had the following JavaScript
+consider the following snippet:
+```js
+function AddTwo(num)
+{
+    return num + 2;
+}
+```
+
+Then, the above function can be invoked from your native add-on using the
+following code:
+```C
+// Get the function named "AddTwo" on the global object
+napi_value global, add_two, arg;
+napi_status status = napi_get_global(env, &global);
+if (status != napi_ok) return;
+
+status = napi_get_named_property(env, global, "AddTwo", &add_two);
+if (status != napi_ok) return;
+
+// var arg = 1337
+status = napi_create_number(env, 1337, &arg);
+if (status != napi_ok) return;
+
+napi_value* argv = &arg;
+size_t argc = 1;
+
+// AddTwo(arg);
+napi_value return_val;
+status = napi_call_function(env, global, add_two, argc, argv, &return_val);
+if (status != napi_ok) return;
+
+// Convert the result back to a native type
+int32_t result;
+status = napi_get_value_int32(env, return_val, &result);
+if (status != napi_ok) return;
+```
+
+### *napi_create_function*
+
+#### Signature
+```C
+napi_status napi_create_function(napi_env env,
+                                 const char* utf8name,
+                                 napi_callback cb,
+                                 void* data,
+                                 napi_value* result);
+```
+
+#### Parameters
+- `[in]  env `: The environment that the API is invoked under
+- `[in]  utf8Name`: The name of the function, encoded as UTF8. This is visible
+within JavaScript as the new function object's `name` property.
+- `[in]  cb`: The native function which should be called when this function 
+object is invoked
+- `[in]  data`: User-provided data context. This will be passed back into the 
+function when invoked later.
+- `[out] result`: `napi_value` representing the JavaScript function object for
+the newly created function
+
+#### Description
+This API allows an add-on author to create a function object in native code. 
+This is the primary mechanism to allow calling *into* the add-on's native code
+*from* Javascript.
+
+Note however, the newly created function is not automatically  visible from
+script after this call. Instead, you must explicitly set a property on the
+Global object (or any other object that is visible to JavaScript), in order for
```

Comment:
Should we be suggesting using the Global object ?  I think we could just leave it as  `set a property on any object that is visible to JavaScript`